src/serialize_sequence.cc

```diff
@@ -88,7 +89,19 @@ void check_stream(std::ostream& stream)
         throw Error(gul14::cat("I/O error: failure on storing step"));
 }
 
-} // namespace anonymous
+} // anonymous namespace
+
+std::string make_sequence_filename(const Sequence& sequence)
+{
+    return make_sequence_filename(sequence.get_label(), sequence.get_unique_id());
+}
+
+std::string make_sequence_filename(gul14::string_view label, const UniqueId& unique_id)
+{
+    return gul14::cat(
+        escape_filename_characters(label), " [",
+        escape_filename_characters(gul14::string_view{ unique_id }), ']');
```

Comment:
```diff
     return gul14::cat(
         escape_filename_characters(label), " [",
-        escape_filename_characters(gul14::string_view{ unique_id }), ']');
+        escape_filename_characters(unique_id.str()), ']');
```

Or rather

```
     return escape_filename_characters(gul14::cat(
         label, " [", unique_id.str(), ']');
```

This is all only needed because you want to avoid a breaking change. But the code would look so much simpler with it, and just adding a tool script to convert old saved sequences to new saved-with-uid sequences would be easy and not much work in the deployment. It can even be called by the debian package installer stuff.

Reply:
The current version is:
```cpp
    return escape_filename_characters(cat(label, " [", unique_id.string(), ']'));
```

I'll close this thread. The discussion about migration scripts or handling old sequences from the library can be continued in another thread if necessary.